src/App.tsx

```diff
@@ -54,48 +64,55 @@ export default function App() {
             getErrorStore().deregister(onErrorChanged)
             getMapOptionsStore().deregister(onMapOptionsChanged)
             getPathDetailsStore().deregister(onPathDetailsChanged)
+            getViewportStore().deregister(onViewportChanged)
         }
     })
 
     const isSmallScreen = useMediaQuery({ query: '(max-width: 44rem)' })
-
-    const [bbox, setBbox] = useState<Bbox>([-180, -90, 180, 90])
     // todo: maybe combine these effects into one? see discussion in #77
     useEffect(() => {
         // make sure the path bbox and the path details bbox take precedence over the info bbox
-        if (!route.selectedPath.bbox && !pathDetails.pathDetailBbox) setBbox(info.bbox)
+        if (!route.selectedPath.bbox && !pathDetails.pathDetailBbox && info.bbox.every(num => num !== 0))
+            Dispatcher.dispatch(new SetViewportToBbox(info.bbox))
```

Comment:
Dispatches should only be caused by external events, such as User Interaction or callbacks issued by the browser. 

This effect is also going to be executed when a store issues a change event caused by another dispatch. This is not supposed to happen. The original Dispatcher implementation of flux forbids this. I guess it is time that we add this to our Dispatcher as well. 

Instead the Viewport Store should listen to the original Actions which may cause changes of the viewport such as RouteReceived or InfoReceived. The store should then decide what the bbox should be and set this in its state. This follows the pattern, that the stores manage the state and the components decide how to represent this state. This pattern is violated here.

Reply:
Ah ok thanks for your explanation, I need to learn more about this then. Should I try to improve this or do you rather do it yourself (and I will be watching you of course :))

Reply:
I can try to do this.

Reply:
Ok, thanks.

Reply:
So the order is:
- 1. pathDetails bbox
- 2. selectedpath bbox
- 3. info bbox
- 4. default bbox

?

Reply:
Yes. But default bbox is just 'world' I think. I have been using this little checklist:

```
app start -> world
/info -> map bbox
/route -> path bbox
clear route -> keep current bbox
change layer -> keep current bbox
select path detail range -> path detail bbox
clear path detail range (click diagram) -> path bbox
```

And I also noticed another bug in master: When we select a path detail range the bbox zooms in, but when we click another alternative in the instructions list then the map does not fly to the other path, probably because the selected path bbox does not update. Not crazy important, but if you change the code here maybe this problem will be solved as well.

Reply:
> And I also noticed another bug in master: When we select a path detail range the bbox zooms in, but when we click another alternative in the instructions list then the map does not fly to the other path, 

Nice, this is also fixed now :+1: